odoo/fields.py

```diff
@@ -705,6 +705,8 @@ def resolve_deps(self, model, path0=[], seen=frozenset()):
                 _logger.warning("Field %s depends on itself; please fix its decorator @api.depends().", self)
             model, path = model0, path0
             for fname in dotnames.split('.'):
+                if fname not in model._fields:
```

Comment:
same as #31889 : instead of an extra lookup in `model._fields`, how about using `model._fields.get(name)` below and raising if the result is `None`? It's not critical but it's not more code, so why make 2 lookups when we can make only 1?

Reply:
because if you have some custom field, `model._fields.[name]` return `None` in some case.
the issue with get(), it return `None` if the key doesn't exist, and return also `None` is the value  is `None`.


maybe me need to write some thing like this:
```python
try:
    field = model._fields[name]
except KeyError:
    raise KeyError("Field : %s, wrong field '%s' of this deps : '%s'." % (self, fname, dotnames)
```

or 

```python
field = model._fields.get(name, 'keyerror_field_doesnt_exist')
if field == 'keyerror_field_doesnt_exist':
    raise KeyError("Field : %s, wrong field '%s' of this deps : '%s'." % (self, fname, dotnames)
```

@odony what is for you the best solution ?

Reply:
ping @odony

Reply:
ping @odony

Reply:
Could you elaborate on your " `model._fields.[name]` return `None` in some case". It sounds odd and maybe there is a bug there

Reply:
I don't remember excatly why it is in some case, it is an issue with custom fields.

Reply:
The dependencies of the custom fields are already tested with the `_check_depends` methods so KeyErrors should not happen on `x_` fields. If we can't reproduce issues (I tried a few), why not applying the suggestion from Olivier and if we realise there is corner cases, we will fix them? Having a field in `self._fields` being stored as `None` seems like a bug that should be fixed there.

Reply:
I have make test on V11 or V12, it is maybe fix in master.
Ok to use the suggestion of Olivier.